lib/bowling.rb

```diff
@@ -0,0 +1,78 @@
+class Bowling
+  attr_reader :bonus, :total, :frame, :current_frame
+
+  def initialize
+    @bonus = 0
+    @total = 0
+    @frame = 1
+    @current_frame = []
+  end
+
+  def add_roll(pins)
+    raise 'The game is over!' unless game_continues?
+
+    bonus? ? score = multiplier(pins) : score = pins
+
+    apply_score(pins, score)
+    end_frame if end_of_frame?
+  end
+
+  private
+
+  def end_frame
+    @bonus += 1 if spare?
+    @bonus += 2 if strike?
+    @current_frame.clear
+    @frame += 1
+  end
+
+  def multiplier(pins)
+    double_bonus? ? pins * 3 : pins * 2
+  end
+
+  def apply_score(pins, score)
+    if double_bonus?
+      @bonus += -2
```

Comment:
rewrite `@bonus -= 2`

Reply:
This really is a bad habit that I picked up in the pre-course, I'll work on that